### PR TITLE
Added equals() and hashCode() to DiscordRichPresence

### PR DESCRIPTION
**Pull request by Almighty-Alpaca**

### CODE REVIEW

File: src/main/java/club/minnced/discord/rpc/DiscordRichPresence.java [hunk 1]
```diff
@@ -178,6 +179,40 @@
...
+                && Objects.equals(partyId, presence.partyId)
+                && Objects.equals(matchSecret, presence.matchSecret)
+                && Objects.equals(joinSecret, presence.joinSecret)
+                && Objects.equals(spectateSecret, presence.spectateSecret);
+    }
+
+    @Override
+    public int hashCode()
+    {
+        return Objects.hash(state, details, startTimestamp, endTimestamp, largeImageKey, largeImageText, smallImageKey,
```
Comment (MinnDevelopment) [inline src/main/java/club/minnced/discord/rpc/DiscordRichPresence.java:210]: I believe the identity hashCode is better for JNA structures as they represent a pointer value

**Review by MinnDevelopment — COMMENTED** [review]
Can you also add equals to the other structures

File: src/main/java/club/minnced/discord/rpc/DiscordRichPresence.java (hunk 1 above)
Comment (Almighty-Alpaca) [reply src/main/java/club/minnced/discord/rpc/DiscordRichPresence.java:210]: That would contradict [the general contract of hashCode()](https://docs.oracle.com/javase/9/docs/api/java/lang/Object.html#hashCode--) though.
> If two objects are equal according to the equals(Object) method, then calling the hashCode method on each of the two objects must produce the same integer result.

Comment (MinnDevelopment) [reply src/main/java/club/minnced/discord/rpc/DiscordRichPresence.java:210]: I see, I still want the other structs to get the same treatment ^^

File: src/main/java/club/minnced/discord/rpc/DiscordEventHandlers.java [hunk 2]
```diff
@@ -105,6 +106,25 @@
      */
     public OnJoinRequest joinRequest;
 
+    @Override
+    public boolean equals(Object o)
+    {
+        if (this == o)
+            return true;
+        if (o == null || getClass() != o.getClass())
```
Comment (MinnDevelopment) [inline src/main/java/club/minnced/discord/rpc/DiscordEventHandlers.java]: What's the reason not to use `instanceof`?

File: src/main/java/club/minnced/discord/rpc/DiscordEventHandlers.java [hunk 3]
```diff
@@ -105,6 +106,25 @@
...
+            return false;
+        if (!super.equals(o))
+            return false;
+        DiscordEventHandlers that = (DiscordEventHandlers) o;
+        return Objects.equals(ready, that.ready) && Objects.equals(disconnected, that.disconnected) && Objects.equals(errored, that.errored) && Objects.equals(joinGame, that.joinGame) && Objects.equals(spectateGame, that.spectateGame) && Objects.equals(joinRequest, that.joinRequest);
```
Comment (MinnDevelopment) [inline src/main/java/club/minnced/discord/rpc/DiscordEventHandlers.java]: Why is this line so long?

**Review by MinnDevelopment — APPROVED** [review]

**Pull request merged**